src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -54,6 +59,33 @@ public OverseasEntitiesService(OverseasEntitySubmissionsRepository overseasEntit
         this.dateTimeNowSupplier = dateTimeNowSupplier;
     }
 
+    public SubmissionType getSubmissionType(String requestId, String overseasEntityId) throws SubmissionNotFoundException {
+        if (!isROEUpdateEnabled) {
+            return SubmissionType.REGISTRATION;
+        }
+        Optional<OverseasEntitySubmissionDto> submissionOpt = getOverseasEntitySubmission(
+                overseasEntityId);
+        if (submissionOpt.isEmpty()) {
+            var errorMessage = "Can not determine submission type";
+            Map<String, Object> logMap = new HashMap<>();
+            logMap.put(OVERSEAS_ENTITY_ID_KEY, overseasEntityId);
+            var error = new SubmissionNotFoundException(errorMessage);
+            ApiLogger.errorContext(requestId, errorMessage, error, logMap);
+            throw error;
```

Comment:
Exception should be created and thrown on the same line, to ensure that line numbers are correct in stack traces.